py/desispec/night_qa.py

```diff
@@ -0,0 +1,740 @@
+#!/usr/bin/env python
+
+# AR general
+import sys
+import os
+from glob import glob
+import tempfile
+import textwrap
+from desiutil.log import get_logger
+# AR scientifical
+import numpy as np
+# AR astropy
+from astropy.table import Table
+from astropy.io import fits
+# AR desispec
+from desispec.fiberbitmasking import get_skysub_fiberbitmask_val
+# AR matplotlib
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib import gridspec
+
+log = get_logger()
+
+def get_nightqa_outfns(outdir, night):
+    """
+    Utility function to get nightqa file names.
+
+    Args:
+        outdir: output folder name (string)
+        night: night (int)
+
+    Returns:
+        dictionary with file names
+    """
+    return {
+        "html" : os.path.join(outdir, "nightqa-{}.html".format(night)),
+        "dark" : os.path.join(outdir, "dark-{}.mp4".format(night)),
+        "badcol" : os.path.join(outdir, "badcol-{}.mp4".format(night)),
+        "ctedet" : os.path.join(outdir, "ctedet-{}.mp4".format(night)),
+        "sframesky" : os.path.join(outdir, "sframesky-{}.mp4".format(night)),
+        "tileqa" : os.path.join(outdir, "tileqa-{}.mp4".format(night)),
+    }
+
+
+
+def get_survey_night_expids(
+    night,
+    survey,
+    datadir = os.getenv("DESI_SPECTRO_DATA")):
```

Comment:
It is generally considered bad python practice to have the parameter defaults call a function in the definition of the function itself, because of possible side effects of running code at the time of evaluating/parsing/loading the module.  Better would be:
```
def get_survey_night_expids(..., datadir=None):
    ...
    if datadir is None:
        datadir = os.getenv("DESI_SPECTRO_DATA") # or desispec.io.rawdata_root()
    ...
```

e.g. if some test code redefined `DESI_SPECTRO_DATA` and then called `get_survey_night_expids`, I think the current code would still use the value of `$DESI_SPECTRO_DATA` at the time the module was loaded instead of the value at the time the function was called.

Reply:
done - thanks for the explanation!